reader_concurrency_semaphore.cc

```diff
@@ -864,35 +864,58 @@ future<> reader_concurrency_semaphore::enqueue_waiter(reader_permit permit, read
 }
 
 void reader_concurrency_semaphore::evict_readers_in_background() {
+    if (_evicting) {
+        return;
+    }
+    _evicting = true;
     // Evict inactive readers in the background while wait list isn't empty
     // This is safe since stop() closes _gate;
     (void)with_gate(_close_readers_gate, [this] {
         return do_until([this] { return _wait_list.empty() || _inactive_reads.empty(); }, [this] {
             return detach_inactive_reader(_inactive_reads.front(), evict_reason::permit).close();
         });
-    });
- }
+    }).finally([this] { _evicting = false; });
+}
+
+reader_concurrency_semaphore::can_admit
+reader_concurrency_semaphore::can_admit_read(reader_permit permit, wait_list_should_be_empty wait_list_empty) const noexcept {
```

Comment:
passing `const reader_permit&` makes more sense IMO to clarify since we don't pass ownership of the permit to the function.

Reply:
Done in v2.